python/dgl/data/qm9.py

```diff
@@ -0,0 +1,180 @@
+"""QM9 dataset for graph property prediction (regression)."""
+import os
+import numpy as np
+import scipy.sparse as sp
+
+from .dgl_dataset import DGLDataset
+from .utils import download, _get_dgl_url
+from ..convert import graph as dgl_graph
+from ..transform import to_bidirected
+from .. import backend as F
+
+class QM9Dataset(DGLDataset):
+    r"""QM9 dataset for graph property prediction (regression)
+
+    This dataset consists of 13,0831 molecules with 12 regression targets.
+    Node means atom and edge means bond.
+
+    Reference: `"Directional Message Passing for Molecular Graphs" <https://arxiv.org/abs/2003.03123>`_
```

Comment:
Maybe use the original reference of QM9 and mention that you use the file processed by D-MPNN.